app/models/user.rb

```diff
@@ -7,17 +7,64 @@ class User < ApplicationRecord
   # end
 
   def starred_repos
-    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    parse(Faraday.get("https://api.github.com/#{self[:username]}/starred?token=#{self.token}"))
   end
 
   def followers
-   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers#{auth}"))
   end
-  
+
   def following
-  parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following#{auth}"))
+  end
+
+  def repos
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/repos#{auth}"))
+  end
+
+  def organizations
+    # connection = Faraday.new(url: "https://api.github.com") do |faraday|
+    #   faraday.params[:access_token] = self.token
+    # end
+    # response = connection.get "#{User.find_by(token: self.token)}/orgs"
+    # # try = parse(response.body)
+    # byebug
+
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/orgs#{auth}"))
   end
 
+
+  def activity
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/events#{auth}"))
+  end
+
+  def activity_of_followed
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/received_events#{auth}"))
+    self.following.each do |f|
+      return parse(Faraday.get("https://api.github.com/users/#{f[:login]}/events#{auth}"))
+    end
+  end
```

Comment:
@AliSchlereth I wasn't sure if it was ok or not that received_events additionally returns the repos that a user followed in addition to the individual users. Currently, this returns the events associated with each user that the current user is following -- am planning on only returning the relevant information in the view, and you cans search through events to find commits by looking for the commit hash (I think!). Quick question - is there a way to "serialize" your PORO objects before they're sent to your views? Let me know if that question doesn't make sense.